tools/integration_tests/log_rotation/log_rotation_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for log rotation of gcsfuse logs.
+
+package log_rotation
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const (
+	testDirName = "LogRotationTest"
+	logFileName = "log.txt"
+	logDirName  = "gcsfuse_integration_test_logs"
+)
+
+var logDirPath string
+var logFilePath string
+
+func getMountConfigForLogRotation(maxFileSizeMB, fileCount int, compress bool,
+	logFilePath string) config.MountConfig {
+	mountConfig := config.MountConfig{
+		LogConfig: config.LogConfig{
+			Severity: config.TRACE,
+			FilePath: logFilePath,
+			LogRotateConfig: config.LogRotateConfig{
+				MaxFileSizeMB: maxFileSizeMB,
+				FileCount:     fileCount,
+				Compress:      compress,
+			},
+		},
+	}
+	return mountConfig
+}
+
+////////////////////////////////////////////////////////////////////////
+// TestMain
+////////////////////////////////////////////////////////////////////////
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	setup.ExitWithFailureIfBothTestBucketAndMountedDirectoryFlagsAreNotSet()
+
+	// Run tests for mountedDirectory only if --mountedDirectory flag is set.
+	logDirPath = path.Join("/tmp", logDirName)
+	logFilePath = path.Join(logDirPath, logFileName)
+	setup.RunTestsForMountedDirectoryFlag(m)
+
+	// Else run tests for testBucket.
+	// Set up test directory.
+	setup.SetUpTestDirForTestBucketFlag()
+
+	logDirPath = setup.SetUpLogDir(logDirName)
```

Comment:
Just curious, for mounted directly (line: 64) we didn't create directory (logDirPath), here we need to create the directory (logDirPath).

Anyway we should create the path using join here only. Then we should create the directory via utility method.

Reply:
The problem here was that log file is generated at the time of mount. So for mounted directory tests, the directory should be there beforehand. SO I've added it to run_tests_mounted_directory script.

I will add a check to fail the test if directory is not present. 

The path is different for both the tests. One is in test directory and the other in /tmp but I will move the logic together.